locked-asset/simple-lock-energy/src/unlock_with_penalty.rs

```diff
@@ -142,7 +142,7 @@ pub trait UnlockWithPenaltyModule:
                 "No tokens remaining after penalty is applied"
             );
 
-            self.burn_penalty(unlocked_token_id, &penalty_amount);
+            self.burn_penalty(unlocked_token_id, payment.token_nonce, &penalty_amount);
```

Comment:
This is wrong. Burned penalty should be of the locked token, not the unlocked now.

Reply:
We should unlock those tokens only which are sent to users - Burning should be done to 50% of the fees. The remaining 50% should be merged to the LKMEX already in storage, thus burning after that the remaining tokens.